mozilla-services/socorro
# Fix bug 1441400: Replace custom exception sanitization processor.

Raven added a configurable processor for sanitizing error data in 6.4.0, which
removes the need for our own custom subclass. In addition, the internal API for
SanitizePasswordsProcessor changed such that our error reporting was broken.

I also added a few extra fields with potentially-sensitive data that we don't really need for error debugging. I'm not actually sure how much we care about CSRF tokens, but I figure it's better to be safe than sorry.

To test this:

1. Acquire a valid Sentry DSN. I used the one from staging. Add it to `my.env` as `RAVEN_DSN`.
2. Checkout the current master branch.
3. Run `docker-compose run webapp ./webapp-django/manage.py raven test`. Your output should look something like this:

   ```
   Starting socorro_memcached_1 ... done
   Starting socorro_elasticsearch_1 ...
   Starting socorro_rabbitmq_1 ...
   Starting socorro_postgresql_1 ...
   Starting socorro_postgresql_1 ... done
   Starting socorro_statsd_1 ... done
   StatsdMetrics configured: statsd:8125 None
   Client configuration:
     base_url       : https://sentry.prod.mozaws.net
     project        : 63
     public_key     : **********************
     secret_key     : **********************

   Sending a test message... Error in sys.excepthook:
   Traceback (most recent call last):
     File "/usr/local/lib/python2.7/site-packages/raven/base.py", line 274, in handle_exception
       self.captureException(exc_info=exc_info, level='fatal')

   <snip>

     File "/app/webapp-django/crashstats/tokens/utils.py", line 9, in RavenSanitizeAuthTokenProcessor
       tuple(SanitizePasswordsProcessor.FIELDS) + ('auth-token',)
   TypeError: 'property' object is not iterable
   ```

4. Check out this branch. Run `docker-compose run webapp ./webapp-django/manage.py raven test` again. This time, the output should show a successful test:

   ```
   Starting socorro_memcached_1 ... done
   Starting socorro_rabbitmq_1 ...
   Starting socorro_postgresql_1 ... done
   Starting socorro_localstack-s3_1 ... done
   StatsdMetrics configured: statsd:8125 None
   Client configuration:
     base_url       : https://sentry.prod.mozaws.net
     project        : 63
     public_key     : ************************
     secret_key     : ************************
   
   Sending a test message... Event ID was 'ea2284f280674fbebac17b5889d7777b'
   Sentry is attempting to send 1 pending error messages
   Waiting up to 10 seconds
   Press Ctrl-C to quit
   ```
5. Check the Sentry instance you sent errors to. You should see an event marked as "Info" called "This is a test message generated using ``raven test``". You may also want to double check the date it was sent in case someone else is also doing some testing.
